hw/vendor/patches/pulp_riscv_dbg/0001-Use-lowrisc-instead-of-PULP-primitives.patch

```diff
@@ -85,81 +98,189 @@ index f742a2b..373f481 100644
  );
  
--  cdc_2phase #(.T(dm::dmi_req_t)) i_cdc_req (
--    .src_rst_ni  ( trst_ni          ),
--    .src_clk_i   ( tck_i            ),
--    .src_data_i  ( jtag_dmi_req_i   ),
--    .src_valid_i ( jtag_dmi_valid_i ),
--    .src_ready_o ( jtag_dmi_ready_o ),
+-  logic                    core_clear_pending;
+-
+-  cdc_2phase_clearable #(.T(dm::dmi_req_t)) i_cdc_req (
+-    .src_rst_ni  ( trst_ni              ),
+-    .src_clear_i ( jtag_dmi_cdc_clear_i ),
+-    .src_clk_i   ( tck_i                ),
+-    .src_clear_pending_o(), // Not used
+-    .src_data_i  ( jtag_dmi_req_i       ),
+-    .src_valid_i ( jtag_dmi_valid_i     ),
+-    .src_ready_o ( jtag_dmi_ready_o     ),
+-
+-    .dst_rst_ni  ( rst_ni               ),
+-    .dst_clear_i ( 1'b0                 ), // No functional reset from core side
+-                                           // used (only async).
+-    .dst_clear_pending_o( core_clear_pending ), // use the clear pending signal
+-                                                // to synchronously clear the
+-                                                // response FIFO in the dm_top
+-                                                // csrs
+-    .dst_clk_i   ( clk_i                ),
+-    .dst_data_o  ( core_dmi_req_o       ),
+-    .dst_valid_o ( core_dmi_valid_o     ),
+-    .dst_ready_i ( core_dmi_ready_i     )
++
 +  // TODO: Make it clean for synthesis.
++  logic jtag_combined_rstn;
++  always_ff @(posedge tck_i or negedge trst_ni) begin
++    if (!trst_ni) begin
++      jtag_combined_rstn <= '0;
++    end else if (jtag_dmi_cdc_clear_i) begin
++      jtag_combined_rstn <= '0;
++    end else begin
++      jtag_combined_rstn <= 1'b1;
++    end
++  end
++
++  logic combined_rstn_premux;
++  prim_flop_2sync #(
++    .Width(1),
++    .ResetValue(0)
++  ) u_combined_rstn_sync (
++    .clk_i,
++    .rst_ni(rst_ni & jtag_combined_rstn),
++    .d_i(1'b1),
++    .q_o(combined_rstn_premux)
+   );
+ 
+-  cdc_2phase_clearable #(.T(dm::dmi_resp_t)) i_cdc_resp (
+-    .src_rst_ni  ( rst_ni               ),
+-    .src_clear_i ( 1'b0                 ), // No functional reset from core side
+-                                           // used (only async ).
+-    .src_clear_pending_o(), // Not used
+-    .src_clk_i   ( clk_i                ),
+-    .src_data_i  ( core_dmi_resp_i      ),
+-    .src_valid_i ( core_dmi_valid_i     ),
+-    .src_ready_o ( core_dmi_ready_o     ),
+-
+-    .dst_rst_ni  ( trst_ni              ),
+-    .dst_clear_i ( jtag_dmi_cdc_clear_i ),
+-    .dst_clear_pending_o(), //Not used
+-    .dst_clk_i   ( tck_i                ),
+-    .dst_data_o  ( jtag_dmi_resp_o      ),
+-    .dst_valid_o ( jtag_dmi_valid_o     ),
+-    .dst_ready_i ( jtag_dmi_ready_i     )
++  logic combined_rstn;
++  prim_clock_mux2 #(
++    .NoFpgaBufG(1'b1)
++  ) u_rst_mux (
++    .clk0_i(combined_rstn_premux),
++    .clk1_i(test_rst_ni),
++    .sel_i(testmode_i),
++    .clk_o(combined_rstn)
+   );
```

Comment:
@arnonsha FYI: this is a new mux that has been inserted to keep the reset signal controllable also in DFT mode.

Reply:
thanks @msfschaffner 
In current design there's no issues with trst in dmi (SpyGlass DFT doesn't issue any errors on this hierarchy).
Is this patch suppose to fix something that is **not yet** in the design ?

Reply:
@arnonsha it fixes #12500 
hopefully also we added all the appropriate dft hooks that nothing should pop on your end either.